third_party/winuwp_h264/H264Encoder/H264Encoder.cc

```diff
@@ -346,10 +356,19 @@ int WinUWPH264EncoderImpl::Encode(
   ComPtr<IMFSample> sample;
   {
     rtc::CritScope lock(&crit_);
-    if (_sampleAttributeQueue.size() > 2) {
-      return WEBRTC_VIDEO_CODEC_OK;
+    // Only encode the frame if the encoder pipeline is not full.
+    if (_sampleAttributeQueue.size() <= 2) {
+      sample = FromVideoFrame(frame);
     }
-    sample = FromVideoFrame(frame);
+  }
+
+  if (!sample) {
+    // Drop the frame. Send a tick to keep the encoder going.
+    lastFrameDropped_ = true;
+    auto timestampHns = GetFrameTimestampHns(frame);
+    ON_SUCCEEDED(sinkWriter_->SendStreamTick(streamIndex_, timestampHns));
```

Comment:
The [docs of `IMFSinkWriter::SendStreamTick()`](https://docs.microsoft.com/en-us/windows/win32/api/mfreadwrite/nf-mfreadwrite-imfsinkwriter-sendstreamtick#remarks) says that we should set the [`MFSampleExtension_Discontinuity` attribute](https://docs.microsoft.com/en-us/windows/win32/medfound/mfsampleextension-discontinuity-attribute) on the first sample after the discontinuity. Do we want to implement that too with the current change?

Reply:
The code is already doing this, see line 314. It's checking lastFrameDropped_ even if that was never set to false before this change 😄

Reply:
I'll put a coin in the "Better GitHub code review" jar. 😒